query/aggregator.go

```diff
@@ -146,7 +160,14 @@ func applyMul(a, b, c *types.Val) error {
 	vBase := getValType(a)
 	switch vBase {
 	case INT:
-		c.Value = a.Value.(int64) * b.Value.(int64)
+		aVal, bVal := a.Value.(int64), b.Value.(int64)
+		c.Value = aVal * bVal
+
+		if aVal == 0 || bVal == 0 || aVal == 1 || bVal == 1 {
+			return nil
+		} else if aVal == math.MinInt64 || bVal == math.MinInt64 || c.Value.(int64)/bVal != aVal {
```

Comment:
Why do we check for `math.MinInt64` here? This doesn't overflow when multiplying by `0` or `1`, for example.

Reply:
0 and 1 are handled in the first if block. Otherwise, it can overflow for any other integer value.